docs/command-line-slangc.md

```diff
@@ -149,6 +149,68 @@ For completeness, here are the options that `slangc` currently accepts:
 
 * `-output-includes`: After pre-processing has been performed will output to via the diagnostics the hierarchy of paths to source files reached 
 
+* -Xname to specify arguments to downstream tool `name` (covered in more detail in "Downstream Arguments")
+
+### Downstream Arguments
+
+During a Slang compilation work may be performed by multiple other stages including downstream compilers and linkers. It isn't possible in general or perhaps even desirable to provide Slang command line equivalents of every option available at every stage of compilation. It is useful to be able to set options specific to a particular compilation stage - to alter code generation, linkage and other options.
+
+The mechanism used here is based on the `-X` mechanism used in GCC, to specify arguments to the linker.
+
+```
+-Xlinker option
+```
+
+When used, option is not interpreted by GCC, but is passed to the linker once compilation is complete. Slang extends this idea in several ways. First there are many more 'downstream' stages available to Slang. These different stages are known as `SlangPassThrough` types in the API and have the following names
+
+* `fxc` - FXC HLSL compiler
+* `dxc` - DXC HLSL compiler
+* `glslang` - GLSLANG GLSL compiler
+* `visualstudio` - Visual Studio C/C++ compiler
+* `clang` - Clang C/C++ compiler
+* `gcc` - GCC C/C++ compiler
+* `genericcpp` - A generic C++ compiler (can be any one of visual studio, clang or gcc depending on system and availability)
```

Comment:
Can we allow `cpp` or `cxx` here as a shorthand?

That said, it would be tricky for somebody to actually form options that any of those compilers would accept.

Also: can the `genericcpp` args go to NVRTC as well? It seems like it accepts many of the same things, right?